projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php

```diff
@@ -139,19 +139,23 @@ public function update_site_options( $request ) {
 		foreach ( $input as $key => $value ) {
 			switch ( $key ) {
 				case 'checklist_statuses':
-					$launchpad_checklist_tasks_statuses_option = (array) get_option( 'launchpad_checklist_tasks_statuses', array() );
-					$launchpad_checklist_tasks_statuses_option = array_merge( $launchpad_checklist_tasks_statuses_option, $value );
-
-					foreach ( $value as $task => $task_value ) {
-						if ( $task_value ) {
-							// If we're marking a task as complete, the value should be truthy, so fire the Tracks event.
-							wpcom_launchpad_track_completed_task( $task );
+					$complete_values   = array();
+					$incomplete_values = array();
+
+					foreach ( $value as $task_id => $task_status ) {
+						if ( $task_status ) {
+							$complete_values[] = $task_id;
+						} else {
+							$incomplete_values[] = $task_id;
 						}
 					}
 
-					if ( update_option( 'launchpad_checklist_tasks_statuses', $launchpad_checklist_tasks_statuses_option ) ) {
+					$updated_complete_tasks   = wpcom_mark_launchpad_task_complete( $complete_values );
+					$updated_incomplete_tasks = wpcom_mark_launchpad_task_incomplete( $incomplete_values );
+					if ( $updated_complete_tasks || $updated_incomplete_tasks ) {
 						$updated[ $key ] = $value;
 					}
+
 					// This will check if we have completed all the tasks and disable Launchpad if so.
 					wpcom_launchpad_checklists()->maybe_disable_launchpad();
```

Comment:
Another side note: I think we should rename this function and the related functions to specify that they are for the fullscreen launchpad: i.e. `maybe_disable_fullscreen_launchpad()`
But that's for another PR.

Reply:
I've created this issue to track this: https://github.com/Automattic/wp-calypso/issues/80140

Reply:
This is already done & merged.